web/views/access.py

```diff
@@ -52,33 +56,41 @@ def get_success_url(self, **kwargs):
         return super().get_success_url()
 
 
-@permission_required(Permissions.EDIT, (Dataset, 'pk', 'dataset_pk'))
-def edit_access(request, pk, dataset_pk):
-    access = get_object_or_404(Access, pk=pk)
-    if request.method == 'POST':
-        form = AccessEditForm(request.POST,  request.FILES, instance=access)
-        if form.is_valid():
-            form.save()
-            messages.add_message(request, messages.SUCCESS, "Access definition updated")
-            redirecturl = reverse_lazy('dataset', kwargs={'pk': dataset_pk})
-            return redirect(to=redirecturl, pk=access.id)
-        else:
-            return JsonResponse(
-                {'error':
-                     {'type': 'Edit error', 'messages': [str(e) for e in form.errors]
-                      }}, status=405)
-    else:
-        access.access_notes = None
-        form = AccessEditForm(instance=access)
-
-    log.debug(submit_url=request.get_full_path())
-    return render(request, 'modal_form.html', {'form': form, 'submit_url': request.get_full_path() })
-
-@require_http_methods(["DELETE"])
+class AccessEditView(CheckerMixin, UpdateView, AjaxViewMixin):
+    model = Access
+    template_name = 'accesses/access_form.html'
+    form_class = AccessEditForm
+    permission_required = Permissions.EDIT
```

Comment:
Is it possible to not require only EDIT permission but also membership in either `local custodians` of the owning dataset or `DataStewardGroup` global group?
The same goes for the AccessCreateView. 

Reasoning behind - we cannot ordinary users grant privilege to create/edit access objects. (local cutodian role is exactly for that - giving notion of responsibility)

Reply:
TODO:
1. For Access itself, add an additional check so that only localcustodians and data-stewards can add/edit accesses
2. To solve in future issue
3. Solved

Reply:
> TODO:
> 
> 1. For Access itself, add an additional check so that only localcustodians and data-stewards can add/edit accesses
> 2. To solve in future issue
> 3. Solved

Yes! :)

Reply:
This should be solved now. Accesses have their own permission checker which returns false if user is not a datasteward or a local custodian